AppCenter/AppCenter/Internals/Channel/MSChannelDefault.m

```diff
@@ -16,8 +17,9 @@ - (instancetype)init {
     _pendingBatchQueueFull = NO;
     _availableBatchFromStorage = NO;
     _enabled = YES;
-
+    _suspended = NO;
     _delegates = [NSHashTable weakObjectsHashTable];
+    _stopFlushingCompletion = [NSMutableArray new];
```

Comment:
What was the motivation behind having an array of completion handlers instead of just a single completion handler?

Reply:
Do you know how a channel can get multiple completion handlers?

Reply:
Pay attention that we immediately stop flushing the channel on processing log if the app is in the background.
Also, see `testStopFlushingWhenItsAlreadyGoing` test.

Reply:
To me the only object the calling the `stopFlushingWithCompletion` in each channels is the logManager, the problem is if two completion handlers are called for the same channel on the logManager then the `remainedChannelsCount` will be corrupted because decrease twice for the same channel.

Reply:
I see, `remainedChannelsCount` is actually increased twice too so not a problem